### DISCUSSION TITLE
Upgrading Next.js 7.0.3 to Latest

### DISCUSSION THREAD

**Discussion by vipindigiqt** [vercel/next.js#40937] [Help] [answered]
Hi there, We have a 4 years old/legacy SaaS application that uses Nextjs 7.0.3 and custom webpack config.

I want to upgrade it to the recent possible next version to get rid of older configs and development experience.

Is there any guide?

I tried to upgrade it to 8.0.0, and I simply got `Internal Server Error`, no helpful errors at all.

**Comment by icyJoseph** [comment, marked as answer]
I don't think there's a good path to jump from 7 to 12. 

From 7 to 8, the upgrading process is apparently fully backward compatible, https://github.com/vercel/next.js/discussions/11878#discussioncomment-4321:

> Next.js 7 => 8 should be fully backwards compatible in most cases. There are no explicit upgrade instructions.
> 
> tl;dr only edge cases should break, which need handled on a case-by-case basis
> 
> Please see the relevant blog post: https://nextjs.org/blog/next-8

From 8 to 9, I guess there's the introduction of data-fetching functions. https://github.com/vercel/next.js/blob/ac31f1636ee15daad983d6d801a1e35764c7fe79/UPGRADING.md a guide I found on the GIT history.

And from there and on, it is mostly that webpack 5 started to be used. And SWC super seeding Babel.

As to why you'd get Internal Server Error, it is anyone's guess, without a repo to look. Perhaps you are using a custom server? 

Glad to help out though, I am just not sure where to start...

I'd say, bump it to `12.3.x`, comment-out/remove all styling or and everything that depends on special WebPack wizardry, double-check your `next.config.js` file,  if you are using a custom server, put it a side for a moment, and try to get the app to run like that, to the bare minimum. 

Probably a good idea to use TypeScript as well.

**Reply by vipindigiqt** [reply]
Hey @icyJoseph, thanks a ton for your time.

Let me try this, I think we are using a custom server for sure.
